blocks/main_loop.py

```diff
@@ -233,8 +234,20 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                callbackname = [key for key, value
+                                in TrainingExtension.__dict__.items()
+                                if getattr(value, '_is_callback', False)]
+                if other not in callbackname:
+                    if isinstance(other, str):
```

Comment:
Oh, I understand now where the confusion comes from: It's okay to raise a `TypeError("{} is not a valid callback.format(other)")` even if `other` is a string. This is because theoretically, the error is that the `str` other can't be cast to a `Callback` type, hence making it a `TypeError` and not a `ValueError`. I guess this kind of depends on your interpretation, but you'll see that e.g. NumPy raises `TypeError`s in these situations as well.